### PR TITLE
User Stories 27-31

### PR DESCRIPTION
**Pull request by WillWeston94**
This is going to be a huge PR

The user stories and the applicable tests were all completed but I have 2 test pending to look into some CSS issues so the coverage is dropping to 95% due to the merchant model and a toggle method in the controller not being "tested"

### CODE REVIEW

File: app/models/merchant.rb [hunk 1]
```diff
@@ -1,5 +1,40 @@
...
+      .where(transactions: { result: 1 })
+      .sum('invoice_items.quantity * invoice_items.unit_price')
+  end
+
+  def top_date
+    invoices
+      .joins(:invoice_items)
+      .group('DATE(invoices.created_at)')
+      .sum('invoice_items.quantity * invoice_items.unit_price')
+      .max_by { |_, revenue| revenue }&.first
```
Comment (NSMcEnroe) [inline app/models/merchant.rb]: I believe max_by is a ruby method, so we may want to refactor this later.

Comment (WillWeston94) [reply app/models/merchant.rb]: it 100% is

**Review by NSMcEnroe — APPROVED** [review]
Thank you for doing the leg work on all of those factory bot objects in the spec file.  I will definitely be stealing those for my tests.  This all looks good, but I'm going to wait and make sure everyone takes a glance at this.

File: app/models/merchant.rb (hunk 1 above)
Comment (WillWeston94) [reply app/models/merchant.rb]: I refactored it to use the Arel.sql

**Review by TheAveryRodriguez — COMMENTED** [review]
Looks great! Good talking through everything on huddle.

**Comment by kicotiel** [comment]
I won't put any specific comments since we covered everything on our group stand up and looked into that Arel thing! 

Routes are like 95% sticking to RESTful (the disable_enable thing I think is the only clear standout, but it can be defended), yay for resources. Model methods look solid and will be used back in US3 as well.

**Review by kicotiel — APPROVED** [review]

**Comment by WillWeston94** [comment]
conflicts resolved for rails helper factory bot config

**Pull request merged**